### DISCUSSION TITLE
Add ability to remove stack from unhandled errors in production

### DISCUSSION THREAD

**Discussion by p-m-p** [remix-run/remix#4564] [Proposals]
## Issue
As per issue #3392 it seems there is no way to omit the stack trace from unhandled errors in loader and action functions. Within the document request the `appState` object contains the serialized error with stack trace and in data request it is in the response body.

## Proposal
Only return the stack property in the serialized error during development. Optionally allow this to be controlled from a configuration option. Possible solution I can add tests for and PR https://github.com/remix-run/remix/compare/main...p-m-p:remix:feature/only-include-error-stack-for-development or happy to take direction and provide an alternative fix.

## References:
[CWE-209](https://cwe.mitre.org/data/definitions/209.html)

**Comment by kiliman** [comment]
Yeah, I have a patch for this. I also include it in my server-side error log handler. I'm using Bugsnag, so both server and client-side errors are sent there. I upload the sourcemaps and then delete them from the server. So I still get a meaningful stacktrace, but the sourcemaps are not sent to the client since they were deleted.

**Comment by brophdawg11** [comment]
This should be resolved as of `1.15.0` via https://github.com/remix-run/remix/pull/5541.  There is also a new [`handleError`](https://remix.run/docs/en/main/file-conventions/entry.server#handleerror) function where you can capture and log/report the error on the server prior to the stack trace being stripped.
